crates/oxc_linter/src/rules/jest/no_hooks.rs

```diff
@@ -0,0 +1,147 @@
+use oxc_span::Span;
+
+use crate::{
+    context::LintContext,
+    rule::Rule,
+    utils::{is_type_of_jest_fn_call, JestFnKind, JestGeneralFnKind, JEST_HOOK_NAMES},
+    AstNode,
+};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("eslint-plugin-jest(no-hooks): ")]
+#[diagnostic(severity(warning))]
+pub struct UnexpectedHookDiagonsitc(#[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct NoHooks {
+    pub allow: Vec<String>,
+}
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// Jest provides global functions for setup and teardown tasks, which are called before/after each test case
+    /// and each test suite. The use of these hooks promotes shared state between tests.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// This rule reports for the following function calls:
+    /// * beforeAll
+    /// * beforeEach
+    /// * afterAll
+    /// * afterEach
+    ///
+    /// ### Example
+    ///
+    /// ```javascript
+    /// function setupFoo(options) { /* ... */ }
+    /// function setupBar(options) { /* ... */ }
+    ///
+    /// describe('foo', () => {
+    ///     let foo;
+    ///     beforeEach(() => {
+    ///         foo = setupFoo();
+    ///     });
+    ///     afterEach(() => {
+    ///         foo = null;
+    ///     });
+    ///     it('does something', () => {
+    ///         expect(foo.doesSomething()).toBe(true);
+    ///     });
+    ///     describe('with bar', () => {
+    ///         let bar;
+    ///         beforeEach(() => {
+    ///             bar = setupBar();
+    ///         });
+    ///         afterEach(() => {
+    ///             bar = null;
+    ///         });
+    ///         it('does something with bar', () => {
+    ///             expect(foo.doesSomething(bar)).toBe(true);
+    ///         });
+    ///     });
+    /// });
+    /// ```
+    NoHooks,
+    style,
+);
+
+impl Rule for NoHooks {
+    fn from_configuration(value: serde_json::Value) -> Self {
+        let allow = value
+            .get(0)
+            .and_then(|config| config.get("allow"))
+            .and_then(serde_json::Value::as_array)
+            .map(|v| {
+                v.iter().filter_map(serde_json::Value::as_str).map(ToString::to_string).collect()
+            })
+            .unwrap_or_default();
+
+        Self { allow }
+    }
+
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        let AstKind::CallExpression(call_expr) = node.kind() else {
+            return;
+        };
+        if !is_type_of_jest_fn_call(
+            call_expr,
+            node,
+            ctx,
+            &[JestFnKind::General(JestGeneralFnKind::Hook)],
+        ) {
+            return;
+        }
+
+        if let Expression::Identifier(ident) = &call_expr.callee {
+            if JEST_HOOK_NAMES.contains(&ident.name.as_str())
+                && !self.allow.contains(&ident.name.to_string())
+            {
+                ctx.diagnostic(UnexpectedHookDiagonsitc(call_expr.span));
```

Comment:
What about we report on `call_expr.callee`? Some people may write many line codes in the hooks function, in our codebase, we prefer it to focus on the error position more detailed, and do you think it will make our find the error easier?